modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogServiceImpl.java

```diff
@@ -130,24 +132,32 @@ public class CatalogServiceImpl extends Producer<CatalogEvent, CatalogEventParam
 
     private final PendingComparableValuesTracker<Integer, Void> versionTracker = new PendingComparableValuesTracker<>(0);
 
-    private final HybridClock clock;
+    private final ClockWaiter clockWaiter;
 
-    private final long delayDurationMs;
+    private final Lazy<Long> delayDurationMs;
```

Comment:
1. This setting must not be changed since it's configured during cluster init (at least, for now; later we might make it changeable)
2. I wanted to avoid going to the configuration on each schema update, but it's actually not a problem as schema updates do not happen too often. So I removed caching of the value.